### ISSUE TITLE
Strange legend orient in facet chart 

### ISSUE THREAD

**Issue by iliatimofeev** [vega/vega-lite#3931]
Please:
- [x] Check for duplicate issues. Please file separate requests as separate issues on GitHub.
related to #2446 
- [x] Describe how to reproduce the bug / the goal of the feature request.
set `"legend": {"orient":"top-right" }` in facet chart
- [x] Use the latest versions of Vega and Vega-Lite.
- [x] Provide an example spec in JSON, wrapped by triple backticks like this:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/seattle-weather.csv"},
  "mark": "bar",
  "encoding": {
    "row": { //<--
      "field": "weather",
      "type": "nominal"
       },
    "x": {
      "timeUnit": "month",
      "field": "date",
      "type": "ordinal"
    },
    "y": {
      "aggregate": "count",
      "type": "quantitative"
    },
    "color": {
      "field": "weather",
      "type": "nominal",
 
      "legend": {
       "orient":"top-right" //<--
      }
    }
  }
}

```

- [x] If applicable, include error messages and screenshots, GIF videos (e.g. using https://www.cockos.com/licecap/), or working examples (e.g. https://bit.ly/vega-lite-blocks)
![export 17](https://user-images.githubusercontent.com/11561581/41922376-a4306a04-796d-11e8-9a8b-62aa046e75af.png)

**Comment by domoritz** [comment]
This is still a problem in the latest master. I wonder whether it has to do with normalization.

**Comment by iliatimofeev** [comment]
Yep, still problem.  VL 3.0.0-rc6 [vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoAmOgCtY3AHaQANOCgATdHkw5gkVgCcx2LTvR9kOvGOQiA7tZbIjdeLAqQAvuoiQMIwBrM0h2dCM1DUhkJXhuTUYlajNQCH8jbgdUsA10qAAzRmQxTVCnHRRIv3yoPABPQmRQpW5UJPRTPPzfbsgAD1Tu-zxGVGQAVSVmUNRlFijawuLS0O08Zpr8yAam0O4jRKVOyG7e7fqhpch0amojYQ3Q+NYlPEXancbmnEgAR1Y6DezB0jCopx6Wyg8TEByunyKJTKvwqLmqwzq3xabQ6pj8GMgdmosWRYDS1wOxTe2B23EIIiMjGo+EguSW3jOGg53iAA)

**Comment by domoritz** [comment]
Hmm, I don't see any problems with the generated spec.

@jheer Could this be a Vega issue or does Vega-Lite need to generate something different in this case.

**Comment by kanitw** [comment]
cc: @jheer
